src/include/catalog/class_handle.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "catalog/catalog.h"
+#include "catalog/catalog_defs.h"
+#include "catalog/catalog_entry.h"
+#include "catalog/table_handle.h"
+#include "storage/sql_table.h"
+#include "transaction/transaction_context.h"
+namespace terrier::catalog {
+
+class Catalog;
+struct SchemaCol;
+
+/**
+ * An ClassEntry is a row in pg_class catalog
+ */
+class ClassEntry : public CatalogEntry<col_oid_t> {
+ public:
+  /**
+   * Constructor
+   * @param oid class def oid
+   * @param sql_table associated with this entry
+   * @param entry a row in pg_class that represents this table
+   */
+  ClassEntry(col_oid_t oid, catalog::SqlTableRW *sql_table, std::vector<type::TransientValue> &&entry)
+      : CatalogEntry(oid, sql_table, std::move(entry)) {}
+};
+
+/**
+ * Class (equiv. of pg_class) stores much of the metadata for
+ * anything that has columns and is like a table.
+ */
+class ClassHandle {
+ public:
+  /**
+   * Get a specific Class entry.
+   * @param txn the transaction that initiates the read
+   * @param oid which entry to return
+   * @return a shared pointer to Class entry;
+   *         NULL if the entry doesn't exist.
+   */
+  std::shared_ptr<ClassEntry> GetClassEntry(transaction::TransactionContext *txn, col_oid_t oid);
+
+  /**
+   * Get a class entry by name
+   * @param txn transaction
+   * @param name to lookup
+   * @return a shared ptr to a Class entry.
+   */
+  std::shared_ptr<ClassEntry> GetClassEntry(transaction::TransactionContext *txn, const char *name);
+
+  /**
+   * Add row into the Class table.
+   * @param txn transaction to run
+   * @param tbl_ptr ptr to the table
+   * @param entry_oid entry oid
+   * @param name class name
+   * @param ns_oid namespace oid
+   * @param ts_oid tablespace oid
+   */
+  void AddEntry(transaction::TransactionContext *txn, int64_t tbl_ptr, int32_t entry_oid, const std::string &name,
+                int32_t ns_oid, int32_t ts_oid);
+
+  /**
+   * Constructor
+   * @param catalog the global catalog object
+   * @param pg_class the pg_class sql table rw helper instance
+   */
+  explicit ClassHandle(Catalog *catalog, std::shared_ptr<catalog::SqlTableRW> pg_class)
+      : catalog_(catalog), pg_class_rw_(std::move(pg_class)) {}
+
+  /**
+   * Create the storage table
+   * @param txn the txn that creates this table
+   * @param catalog ptr to the catalog
+   * @param db_oid db_oid of this handle
+   * @param name catalog name
+   * @return a shared pointer to the catalog table
+   */
+  static std::shared_ptr<catalog::SqlTableRW> Create(transaction::TransactionContext *txn, Catalog *catalog,
```

Comment:
Why does this need to be static? We should probably move into a helper class.

Reply:
It is static because:
- The Create function needs to be called before an instance of the class can be instantiated. 
- To keep the code related to the class in one place. The first implementation had creation and column definitions in the catalog class code. This was hard to read and poor for maintainability.

This could be moved to a util class, but IMO clearer to keep all the related code together. Singletons are not being created.